sdv/multi_table/base.py

```diff
@@ -201,30 +183,9 @@ def validate(self, data):
                     * values of a column don't satisfy their sdtype
         """
         errors = []
-        missing_tables = set(self.metadata.tables) - set(data)
-        if missing_tables:
-            errors.append(f'The provided data is missing the tables {missing_tables}.')
-
-        for table_name, table_data in data.items():
-            try:
-                self._table_synthesizers[table_name].validate(table_data)
-
-            except InvalidDataError as error:
-                error_msg = f"Table: '{table_name}'"
-                for _error in error.errors:
-                    error_msg += f'\nError: {_error}'
-
-                errors.append(error_msg)
-
-            except ValueError as error:
-                errors.append(str(error))
-
-            except KeyError:
-                continue
-
-        foreign_key_errors = self._validate_foreign_keys(data)
-        if foreign_key_errors:
-            errors.append(foreign_key_errors)
+        errors += self.metadata._validate_missing_tables(data)
+        errors += self._validate_all_tables(data)
+        errors += self.metadata._validate_foreign_keys(data)
```

Comment:
alternatively you could call `self.metadata.validate` and then make a separate method that loops through the synthesizers and calls the constraint validation as well as the synthesizer specific validation